### ISSUE TITLE
`no-duplicates` false positive regression

### ISSUE THREAD

**Issue by LukeAbby** [un-ts/eslint-plugin-import-x#398]
When updating to 4.6.2 I have a new false positive for `import-x/no-duplicates`. It requires a fairly involved structure with a circular loop of imports so may have the same underlying issue as #397.

Here's the reproduction zip:
[reproduction.zip](https://github.com/user-attachments/files/20917509/reproduction.zip)

1. Unextract the reproduction.
2. Open terminal in folder.
3. Run `npm install`
4. Run `npx eslint` and you'll get a false positive for `import-x/no-duplicates`.

**Comment by JounQin** [comment]
Please use a GitHub repo as reproduction instead.

**Comment by LukeAbby** [comment]
I delete inactive repos so I prefer zips to avoid bitrot in issues. Nevertheless here you go: https://github.com/LukeAbby/no-duplicates-reproduction

**Comment by JounQin** [comment]
Please help to test:

```sh
npm i https://pkg.pr.new/eslint-plugin-import-x@399
```

**Comment by LukeAbby** [comment]
That fixed it! Thanks!
